### ISSUE TITLE
how (if possible already) to communicate exceptions raised in callbacks?

### ISSUE THREAD

**Issue by yarikoptic** [pyout/pyout#94]
With following example:

```shell
$> cat /tmp/pyoutout.py
import pyout

def cb():
    yield "yielding"
    yield "will raise now"
    raise Exception("A very important error")

with pyout.Tabular(style={"header_": dict(bold=True,)}) as out:
   out({'path': 'mypath',
        'cb': cb})
```
I get just
```shell
$> python /tmp/pyoutout.py
path   cb            
mypath will raise now
```
with normal exit code.
I guess ATM there is no provisioning for handling exceptions thrown in callbacks.
One of the solutions could be adding an `exceptions_callback` option to `Tabular` which is called with the caught exception and the record/row on which it happened.

It is of a low priority since workaround could be to guard against all exceptions within callbacks themselves and somehow communicate back (I use `message` column) that something abnormal happened

**Comment by kyleam** [comment]
> how (if possible already) to communicate exceptions raised in callbacks?

pyout uses stdlib's `multiprocessing.dummy.Pool` underneath, and I'm not aware of a way to catch an exception raised in a child thread.

While we could find some way to communicate that an exception happened, it seems like it'd be best to switch away from `multiprocessing.dummy`.  `concurrent.futures`, for example, seems to handle the exceptions nicely.

```python
from concurrent.futures import ThreadPoolExecutor

def fn():
    raise ValueError("blah")

with ThreadPoolExecutor() as executor:
    future = executor.submit(fn)
    future.result()
```

```
Traceback (most recent call last):
  File "/tmp/confut.py", line 10, in <module>
    future.result()
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/tmp/confut.py", line 5, in fn
    raise ValueError("blah")
ValueError: blah
```

That's available as of Python 3.2.  It's been backported for Python 2, but  I'd say pyout should just drop Python 2 support given that any libraries you want to use pyout in have dropped Python 2 support, at least in their development line.

**Comment by yarikoptic** [comment]
Yes, dropping 2 sounds like a timely thing to do
